src/frontend/src/binder/mod.rs

```diff
@@ -90,10 +93,78 @@ pub struct Binder {
 
     /// `ShareId`s identifying shared views.
     shared_views: HashMap<ViewId, ShareId>,
+
+    param_types: ParameterTypes,
+}
+
+#[derive(Clone, Debug)]
+pub struct ParameterTypes(Arc<RwLock<HashMap<u64, Option<DataType>>>>);
+
+impl ParameterTypes {
+    pub fn new(param_types: Vec<DataType>) -> Self {
```

Comment:
Is this `param_types: Vec<DataType>` going to be the explicitly specified types from pgwire? Let's call it `specified_types` or `initial_types`, or write some comments about it.